pkg/api/kubernetesagentpool/validate.go

```diff
@@ -0,0 +1,99 @@
+package kubernetesagentpool
+
+import (
+	"fmt"
+	"strings"
+)
+
+func (p *AgentPool) Validate() error {
+
+	// -------------------------------
+	// Kubernetes Version
+	if p.Properties.KubernetesVersion == "" {
+		return fmt.Errorf("Empty Kubernetes version")
+	}
+	if !strings.Contains(p.Properties.KubernetesVersion, ".") {
+		return fmt.Errorf("Invalid Kubernetes version")
+	}
+
+	// -------------------------------
+	// Kubernetes Version
+	if p.Properties.KubernetesEndpoint == "" {
+		return fmt.Errorf("Empty Kubernetes endpoint")
+	}
+	if !strings.Contains(p.Properties.KubernetesVersion, ".") {
+		return fmt.Errorf("Invalid Kubernetes endpoint")
+	}
+
+	// -------------------------------
+	// Agent pools
+	if len(p.Properties.AgentPoolProfiles) < 1 {
+		return fmt.Errorf("Missing agent pool definition")
+	}
+	for _, agentPool := range p.Properties.AgentPoolProfiles {
+		if agentPool.Count < 1 {
+			return fmt.Errorf("Must using a value greater than 0 for agent pool count")
+		}
+		if agentPool.OSDiskSizeGb != 0 && agentPool.OSDiskSizeGb < 2 {
```

Comment:
Why the `!=0` check? Additionally... if I set disk size == 2, this check would pass without error. Error message should therefore probably read:

"Must use a disk size greater than or equal to 2gb for agent pool"